File: lang/go/test/muse/data_structures/LinkedQueue/LinkedQueueTest.go
package main

import (
    "muse/data_structures/LinkedQueue"
    "muse/util/TestRunner"
)

func testLinkedQueue() bool {
    size := 8192

    queue := LinkedQueue.New()

    for i := 1; i <= size; i++ {
        queue.Enqueue(i)
    }

    if queue.Size() != size {
        return false
    }

    for i := 1; i <= size; i++ {
        if queue.Peek() != i {
            return false
        }
        queue.Dequeue()
    }

    return queue.IsEmpty()
}

func main() {
    TestRunner.ParseTest(testLinkedQueue())
}

File: archive/ruby/muse/data_structures/LinkedQueueTest.rb
require "muse/data_structures/LinkedQueue"
require "muse/util/TestRunner"

def testLinkedQueue
  size = 8192

  queue = LinkedQueue::LinkedQueue.new

  1.upto(size) do |i|
    queue.enqueue(i)
  end

  return false if queue.size != size

  1.upto(size) do |i|
    return false if queue.peek != i

    queue.dequeue
  end

  queue.isEmpty
end

TestRunner.pick(testLinkedQueue) if __FILE__ == $PROGRAM_NAME
